### PR TITLE
[Feature/metadata check] Metadata check and db integration

### PR DESCRIPTION
**Pull request by PascalDR**
This PR is relative to issue [112](https://github.com/italia/eudi-wallet-it-python/issues/112) and resolve todo 177 and 178.
It integrates the db operations for save the metadata and fixes some issuer with TrustedEvaluationHelper.

### CODE REVIEW

File: pyeudiw/storage/mongo_storage.py [hunk 1]
```diff
@@ -270,6 +271,18 @@ def add_trust_attestation(self, entity_id: str, attestation: list[str], exp: dat
...
 
+    def add_trust_attestation_metadata(self, entity_id: str, metadata: dict):
+        entity = self._get_trust_attestation("trust_attestations", entity_id)
+
+        if entity is None:
+            raise ValueError(
+                f'Document with entity_id {entity_id} not found.'
+            )
+        
+        entity["metadata"] = metadata
```
Comment (peppelinux) [inline pyeudiw/storage/mongo_storage.py:282]: as said yesterday

entity['metadata'][$metadata_type] :dict

where $metadata_type is set of: openid_credential_issuer, wallet_provider, wallet_relying_party and so on ...

**Review by peppelinux — CHANGES_REQUESTED** [review]

File: pyeudiw/tests/storage/test_db_engine.py [hunk 2]
```diff
@@ -131,6 +131,25 @@ def test_update_unexistent_trusted_attestation(self):
...
+        
+        ta = self.engine.get_trust_attestation(self.federation_entity_id)
+
+        assert ta.get("metadata", None) != None
+        assert ta["metadata"] == {"metadata": "test"}
+
+    def test_update_unexistent_trusted_attestation_metadata(self):
+        try:
+            self.engine.add_trust_attestation_metadata(
+                "test", {"metadata": "test"})
```
Comment (peppelinux) [inline pyeudiw/tests/storage/test_db_engine.py:148]: see my prev comment

File: pyeudiw/tests/trust/test_TrustEvaluationHelper.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import pytest
```
Comment (peppelinux) [inline pyeudiw/tests/trust/test_TrustEvaluationHelper.py:1]: please, the name of this file must be lowercased and snake typed

this seems capitalized and camel typed

File: pyeudiw/trust/__init__.py [hunk 4]
```diff
@@ -13,6 +13,9 @@
     MissingTrustType
 )
 
+import pyeudiw.metadata.policy as pcl
+
+#from pyeudiw.metadata.policy import TrustChainPolicy, combine
```
Comment (peppelinux) [inline pyeudiw/trust/__init__.py:18]: remove this comment if not of interest anymore

File: pyeudiw/trust/__init__.py [hunk 5]
```diff
@@ -127,9 +140,12 @@ def _handle_x509_pem(self):
                 "a recognizable Trust Anchor."
             )
 
-        pem = trust_anchor['x509']['pem']
-
-        _is_valid = verify_x509_anchor(pem)
+        try:
+            pem = trust_anchor['x509']['pem']
+            _is_valid = verify_x509_anchor(pem)
+        except KeyError:
```
Comment (peppelinux) [inline pyeudiw/trust/__init__.py:146]: what if the exception is raised by verify_x509_anchor?
except KeyError is not sufficient

File: pyeudiw/trust/__init__.py [hunk 6]
```diff
@@ -127,9 +140,12 @@ def _handle_x509_pem(self):
...
+            raise MissingTrustType(
+                f"Trust Anchor: '{trust_anchor_eid}' has no x509 trusst entity")
```
Comment (peppelinux) [inline pyeudiw/trust/__init__.py:148]: ss -> s

**Review by peppelinux — CHANGES_REQUESTED** [review]
it seems good, however there are still few things to improve before the approvals

File: pyeudiw/storage/mongo_storage.py (hunk 1 above)
Comment (PascalDR) [reply pyeudiw/storage/mongo_storage.py]: Resolved in [1d3fa1b](https://github.com/italia/eudi-wallet-it-python/pull/188/commits/1d3fa1b74b7251c4f4702dabb7aedcc984824572)

File: pyeudiw/tests/storage/test_db_engine.py [hunk 7]
```diff
@@ -131,6 +131,25 @@ def test_update_unexistent_trusted_attestation(self):
...
+    def test_update_trusted_attestation_metadata(self):
+        replica_count = self.engine.add_trust_attestation_metadata(
+            self.federation_entity_id, {"metadata": "test"})
+        
+        assert replica_count > 0
+        
+        ta = self.engine.get_trust_attestation(self.federation_entity_id)
+
+        assert ta.get("metadata", None) != None
+        assert ta["metadata"] == {"metadata": "test"}
```
[the remark this answers is not in the document]
Comment (PascalDR) [reply pyeudiw/tests/storage/test_db_engine.py]: Resolved in [d90974c](https://github.com/italia/eudi-wallet-it-python/pull/188/commits/d90974ce11036794799a476e4307cfd2796149e8)

File: pyeudiw/tests/storage/test_db_engine.py (hunk 2 above)
Comment (PascalDR) [reply pyeudiw/tests/storage/test_db_engine.py]: Resolved in [d90974c](https://github.com/italia/eudi-wallet-it-python/pull/188/commits/d90974ce11036794799a476e4307cfd2796149e8)

File: pyeudiw/tests/trust/test_TrustEvaluationHelper.py (hunk 3 above)
Comment (PascalDR) [reply pyeudiw/tests/trust/test_TrustEvaluationHelper.py]: Resolved [850d432](https://github.com/italia/eudi-wallet-it-python/pull/188/commits/850d432f9239edfa89a5ab2c2f9d0ea4547a50fc)

File: pyeudiw/trust/__init__.py (hunk 4 above)
Comment (PascalDR) [reply pyeudiw/trust/__init__.py]: Resolved in [c062b35](https://github.com/italia/eudi-wallet-it-python/pull/188/commits/c062b354e94337a19c652e5fdc50240b3a8422f1)

File: pyeudiw/trust/__init__.py (hunk 5 above)
Comment (PascalDR) [reply pyeudiw/trust/__init__.py]: Resolved in [fix: fixed x509 verification exception handling](https://github.com/italia/eudi-wallet-it-python/pull/188/commits/bf3843cf5708b8826861ae968add41388d6fe0e9)

File: pyeudiw/trust/__init__.py (hunk 6 above)
Comment (PascalDR) [reply pyeudiw/trust/__init__.py]: Resolved in [5a74ea0](https://github.com/italia/eudi-wallet-it-python/pull/188/commits/5a74ea0a4091d74e13d3975b140767d8b65ab94f)

**Review by peppelinux — APPROVED** [review]

**Pull request merged**